ot/mpz-ot-core/src/ferret/utils.rs

```diff
@@ -0,0 +1,213 @@
+//! Utils for the implementation of Ferret.
+
+use mpz_core::{aes::AesEncryptor, Block};
+
+use super::{CUCKOO_HASH_NUM, CUCKOO_TRIAL_NUM};
+
+/// Errors that can occur when inserting Cuckoo hash.
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum CuckooHashError {
+    #[error("invalid Cuckoo hash state: expected {0}")]
+    CuckooHashLoop(String),
+}
```

Comment:
We can simplify this as I believe there is only going to be 1 error case:

```suggestion
/// Cuckoo hash eviction error
#[derive(Debug, thiserror::Error)]
#[error("item evicted from table")]
pub struct Eviction;
```